pkgs/development/beam-modules/build-mix.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, writeText, elixir, erlang, hex, lib }:
+
+{ name
+, version
+, src
+, buildInputs ? [ ]
+, beamDeps ? [ ]
+, propagatedBuildInputs ? [ ]
+, postPatch ? ""
+, compilePorts ? false
+, meta ? { }
+, enableDebugInfo ? false
+, mixEnv ? "prod"
+, ...
+}@attrs:
+
+with lib;
+let
+  shell = drv: stdenv.mkDerivation {
+    name = "interactive-shell-${drv.name}";
+    buildInputs = [ drv ];
+  };
+
+  pkg = self: stdenv.mkDerivation (attrs // {
+    name = "${name}-${version}";
+    inherit version;
+    inherit src;
+
+    MIX_ENV = mixEnv;
+    MIX_DEBUG = if enableDebugInfo then 1 else 0;
+    HEX_OFFLINE = 1;
+
+    setupHook = attrs.setupHook or
+      writeText "setupHook.sh" ''
+      addToSearchPath ERL_LIBS "$1/lib/erlang/lib"
+    '';
+
+    buildInputs = buildInputs ++ [ elixir hex ];
+    propagatedBuildInputs = propagatedBuildInputs ++ beamDeps;
+
+    configurePhase = attrs.configurePhase or ''
+      runHook preConfigure
+
+      mkdir -p _build/$MIX_ENV/lib
```

Comment:
My understanding is the compiled modules use $ERL_LIBS to find imports at runtime, but mix does not. Mix expects everything to be in the _build directory.

Maybe everything worked without the configurePhase because the dependencies weren't actually required to build the project. You still want to list them so they will be installed and added to ERL_LIBS, which will be needed at runtime.

Reply:
I think you're right here, we shouldn't rely on this behavior if it's not documented anywhere.
I've tried to look for reference to the ERL_LIBS in the mix source code but could only find this one
https://github.com/elixir-lang/elixir/blob/78732f428f78f0031ac4e2c6c144f409e4da1d64/lib/mix/lib/mix/tasks/deps.compile.ex#L310
And it doesn't seem that meaningful.
On thee other hand, I couldn't find any mention on the mix behavior with regards to the build directory.
Let me ask on the elixir mailing list so we can try to get a clear answer.

Reply:
Feel free to add precisions if you feel I missed something
https://groups.google.com/g/elixir-lang-core/c/NnfmSnnuQ10

Reply:
After doing it the hard way, I confirmed the easy way: https://github.com/elixir-lang/elixir/blob/78732f428f78f0031ac4e2c6c144f409e4da1d64/lib/mix/lib/mix/tasks/deps.compile.ex#L310

Mix does use ERL_LIBS when compiling. And has for about 6 years: https://github.com/elixir-lang/elixir/commit/c7e9e4f304264c8fc47b799d672a4b0b6fadd7b9

I think we can completely remove the configurePhase.